packages/components/src/components/domainproperties/dataset/DatasetPropertiesPanelFormElements.tsx

```diff
@@ -0,0 +1,297 @@
+        const value = description === null ? '' : description;
+
+        return(
+            <Row className={'margin-top'}>
+                <Col xs={4}>
+                    Description
+                </Col>
+
+                <Col xs={7}>
+                    <textarea
+                        className="form-control textarea-noresize"
+                        id="description"
+                        value={value}
+                        onChange={onInputChange}
+                    />
+                </Col>
+            </Row>
+        );
+    }
+}
+
+export class SelectPropertyInput extends React.PureComponent<any> {
+    render() {
+        const { label, labelKey, valueKey, onInputChange, options, disabled } = this.props;
+
+        return(
+            <Row className={'margin-top'}>
+                <Col xs={4} >
+                    {label}
+                </Col>
+
+                <Col xs={7} >
+                    <SelectInput
+                        onChange={onInputChange}
+                        options={options}
+                        inputClass="" // This attr is necessary for proper styling
+                        labelClass=""
+                        valueKey={valueKey}
+                        labelKey={labelKey}
+                        formsy={false}
+                        multiple={false}
+                        required={false}
+                        disabled={disabled}
+                    />
+                </Col>
+
+                <Col lg={3}/>
+            </Row>
+        );
+    }
+}
+
+export class BasicPropertiesFields extends React.PureComponent<BasicPropertiesInputsProps, any> {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+             category: this.props.model.categoryId,
+             availableCategories: []
+        };
+    }
+
+    componentDidMount() {
+        const { model } = this.props;
+
+        // Ajax call handling to get available categories
+        fetchCategories()
+            .then((data) => {
+                this.setState(() => ({
+                    availableCategories: data.categories
+                }))
+            })
+    }
+
+    render() {
+        const { name, description, categoryId, label, onInputChange, onCategoryChange } = this.props;
+        const { availableCategories } = this.state;
+
+        const nameTip =
+            <>
+                Required. This name must be unique. It is used when identifying datasets during data upload.
+            </> as JSX.Element;
+
+        const categoryTip =
+            <>
+                Assigning a category to a dataset will group it with similar datasets in the navigator and data browser.
+            </> as JSX.Element;
+
+        const labelTip =
+            <>
+                The name of the dataset shown to users. If no Label is provided, the Name is used.
+            </> as JSX.Element;
+
+        return (
+            <Col md={6}>
```

Comment:
I'd probably move this <Col> up one level so that the parent handles the layout of the <Row> there.